commit 9011ff6c9fb5128dd50b4210e05199ea6337b82b
Author: Willy Tarreau <w@1wt.eu>
Date:   Sat Oct 20 17:45:48 2018 +0200

    BUILD: compiler: rename __unreachable() to my_unreachable()
    
    Olivier reported that on FreeBSD __unreachable is already defined
    and causes build warnings. Let's rename it then.
    
    (cherry picked from commit 4e7cc3381b27e3971b02b73a113ecc13916e1f20)
    Signed-off-by: Willy Tarreau <w@1wt.eu>

diff --git a/include/common/compiler.h b/include/common/compiler.h
index 6f4f5a67..60549307 100644
--- a/include/common/compiler.h
+++ b/include/common/compiler.h
@@ -89,9 +89,9 @@
  * below was introduced in gcc 4.5, and before it we didn't care.
  */
 #if __GNUC__ >= 5 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)
-#define __unreachable() __builtin_unreachable()
+#define my_unreachable() __builtin_unreachable()
 #else
-#define __unreachable()
+#define my_unreachable()
 #endif
 
 /*
diff --git a/src/hlua.c b/src/hlua.c
index c3bb269a..085544dc 100644
--- a/src/hlua.c
+++ b/src/hlua.c
@@ -64,7 +64,7 @@
  *   MAY_LJMP() marks an lua function that may use longjmp.
  */
 #define __LJMP
-#define WILL_LJMP(func) do { func; __unreachable(); } while(0)
+#define WILL_LJMP(func) do { func; my_unreachable(); } while(0)
 #define MAY_LJMP(func) func
 
 /* This couple of function executes securely some Lua calls outside of
